File: java/src/main/java/de/tschuehly/microstreamtest/persistence/PersonRepository.java
package de.tschuehly.microstreamtest.persistence;

import java.util.List;
import org.springframework.stereotype.Repository;

@Repository
public class PersonRepository extends CrudRepository<Person> {

  public PersonRepository(Root root) {
    super(root, root.persons);
  }

  public List<Person> findByName(String name){
    return readList(() -> root.persons.values().stream().filter(person -> person.name.equals(name)).toList());
  }
}

File: src/main/kotlin/de/tschuehly/microstreamtest/persistence/PersonRepository.kt
package de.tschuehly.microstreamtest.persistence

import org.springframework.stereotype.Repository

@Repository
class PersonRepository(
    override val root: Root
) : CrudRepository<Person>(root, root.personMap){
    fun findByName(name: String): List<Person> {
        return root.personMap.values.filter {
            it.name == name
        }
    }
}